codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -402,20 +392,20 @@ private void verifyUnknownFrame() throws Http2Exception {
         verifyNotProcessingHeaders();
     }
 
-    private void readDataFrame(ChannelHandlerContext ctx, ByteBuf payload, int payloadEndIndex,
+    private void readDataFrame(ChannelHandlerContext ctx, ByteBuf payload,
             Http2FrameListener listener) throws Http2Exception {
         int padding = readPadding(payload);
         verifyPadding(padding);
 
         // Determine how much data there is to read by removing the trailing
         // padding.
-        int dataLength = lengthWithoutTrailingPadding(payloadEndIndex - payload.readerIndex(), padding);
+        int dataLength = lengthWithoutTrailingPadding(payloadLength - payload.readerIndex(), padding);
```

Comment:
I think these can be even simpler: 
```suggestion
        int dataLength = lengthWithoutTrailingPadding(payload.readableBytes(), padding);
```

Reply:
Agree, I'll make the changes to all of these